lib/src/test/groovy/com/cloudbees/groovy/cps/CpsTransformerTest.groovy

```diff
@@ -186,7 +186,7 @@ class CpsTransformerTest extends AbstractGroovyCpsTest {
             } catch(StringIndexOutOfBoundsException e) {
               return e.message;
             }
-        """)=="String index out of range: -2"
+        """)=="begin 5, end 3, length 3"
```

Comment:
The message changed in Java 9.

Reply:
Actually, for this specific overload of `substring`, [this commit](https://github.com/openjdk/jdk17/commit/c0b8844dce2335531d47eeaae8f6de3a48c5c2b6) which appears to have first landed in JDK 14 is what changes the message.

That makes things a bit awkward, since we want to be able to build using at least Java 11 and Java 17, but we can adapt the test to support either version.